test/cpp/jit/test_custom_operators.h

```diff
@@ -207,6 +210,53 @@ void testCustomOperators() {
     tracer::abandon();
   }
 }
+
+void testCustomOperatorAliasing() {
+  RegisterOperators reg({createOperator(
+      "foo::aliasing", [](at::Tensor a, at::Tensor b) -> at::Tensor {
+        a.add_(b);
+        return a;
+      })});
+  auto& ops = getAllOperatorsFor(Symbol::fromQualString("foo::aliasing"));
+  auto& op = ops.front();
+
+  ASSERT_TRUE(op->schema().has_alias_annotation());
+
+  {
+    auto graph = std::make_shared<Graph>();
+    script::parseIR(
+        R"IR(
+graph(%x: Tensor, %y: Tensor):
+  %ret : Tensor = foo::aliasing(%x, %y)
+  return (%ret)
+  )IR",
+        graph.get());
+
+    auto opNode = *graph->block()->nodes().begin();
+
+    AliasDb aliasDb(graph);
+    for (const auto input : opNode->inputs()) {
+      // The custom op writes to all its inputs
+      ASSERT_TRUE(aliasDb.writesToAlias(opNode, {input}));
+      // The output should be a wildcard and thus alias all inputs
+      ASSERT_TRUE(aliasDb.mayAlias(opNode->output(), input));
+    }
+  }
+  {
+    // DCE should not remove a custom op
+    auto graph = std::make_shared<Graph>();
+    const auto text = R"IR(
+graph(%x: Tensor, %y: Tensor):
+  # CHECK: foo::aliasing
+  %ret : Tensor = foo::aliasing(%x, %y)
```

Comment:
Can you add a test for when there are no inputs or outputs, to test the hasSideEffects() case in addition to the case where we assume inputs are written to?

Reply:
Apparently the schema inference breaks if you try to return void?

Reply:
Could also add a function with no inputs and an unused int output to test the same condition